src/domain/utils/optionsHelper.ts

```diff
@@ -0,0 +1,264 @@
+export type OptionType = {
+    label: string;
+    isHidden?: boolean;
+};
+
+export const PPSSurveyNationalOptions = (hasReadAccess: boolean, hasCaptureAccess: boolean) => {
```

Comment:
This file has functions with repeated functionality . Typically, we'd DRY it with a helper function and another function which takes the survey type as argument calls the helper function and conditionally returns different options based on form type.

Reply:
If I understand correctly I guess that we are already doing it on the getSurveyOptions no? Conditionally returning the options based on the surveyType there. Here it's building the options for each type.

Reply:
if you look at this file, there is repeated code, all Edits are hidden if there is hasReadAccess, all view are hidden if there is hasCaptureAccess , all delete are hidden if hasReadAccess . DRY - Do not Repeat Yourself. Build the options based on the survey form type, and return the common functionality form a base function. This significantly reduce the LOC.

Reply:
Ok, there already is a default option with only edit, view and delete, I'll use it on all the other functions and those 3 common ones are not repeated anymore. But each function in this file is only about one form type, I could do it all in getSurveyOptions, since there is where it is conditionally checking the form types, but it would cluster that function. That's why based on each formType, I build a function that returns the options for each one.

But if the menu items order is important, for example delete has to be last, I think repeating menu items is a better alternative than adding sorting, since this is just basically a json with conditions.